docs/_pages/walk-through.md

```diff
@@ -0,0 +1,578 @@
+```js
+perform(callback) {
+    driver.findElement(By.id('article1'))
+    .click()
+    .then(callback, callback);
+},
+```
+Once the preconditions are met, you can execute your perform block, in this case we are simply making a selenium call, which finds the element by its ID, then clicks on the element. Afterwhich it resolves or rejects the promise. If any of this language sounds unfamiliar, you may want to revisit some of the above posted selenium documentation or take a look at
+[promises documentation.](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise)
+```js
+effects(expectedState) {
+```
+The effects are going to be a little more split up, because there is a lot to go over. This first part is just passing in the expected state to the effects function which was covered earlier. 
+```js
+expectedState.stash();
+```
+Once you have the expected state, you're going to want to stash the expected state as we are not navigating away from the page, but rather adding a component on top of it. Stashing is similar to clearing the state, but if you plan to return to the previous state, it is better to hold onto it so re-navigating to the previous state only requires a pop as opposed to rebuilding the state from a cleared state. 
+```js
+expectedState.createAndAddComponent({
+    type: 'ViewStoryModal',
+    name: 'article1ViewModal',
+```
+Here we are creating and adding a component to the expected, as we did before with adding the mainSiteLayout, we will be adding the ViewStoryModal component, whcih as you might have guessed displays the storymodal. As this action is particular to article1, we will include it as the name of the view modal. the final thing we are sending is the state of the component we are adding, which is detailed below.
+```js
+state: {
+    displayed: true,
+    modalTitle: {
+        displayed: true,
+        text: 'Test Article One',
+    },
+    modalBodyText: {
+        displayed: true,
+        text: 'This is the body text of a first test article that is long enough to pass by the preview and still show more in the pop up modal.',
+    },
+    closeButton: {
+        displayed: true,
+    },
+},
+```
+
+All of the pieces above in the state should look familiar as they were added the same way within the effects of the navigation component. They simply detail what to expect in the component being added. Below is something slightly different, the options are just variable parameters to except within the component being added, this is useful when referencing parent values within a model that inherits certain properties from the parent, such as an ID appended to additional information. 
```

Comment:
The explanations of options here needs a bit more clarity or simplification. Perhaps "options can be seen as simply parameters to the component" or something along those lines.